site/cypress/e2e/cypress-feature/late_days_cache.spec.js

```diff
@@ -0,0 +1,349 @@
+    cy.login('instructor');
+    cy.visit(['sample', 'bulk_late_days']);
+    for (const user_id of all_user_ids) {
+        // Gradeable # of late days used should be empty
+        cy.get(`[USER_ID="${user_id}"] > [id="Late Allowed Homework"]`)
+            .then((cell) => expect(cell.text().trim()).to.equal(''));
+
+        // Remaining late days isnt known, should be empty
+        cy.get(`[USER_ID="${user_id}"] > [ld_id="Late Days Remaining"]`)
+            .then((cell) => expect(cell.text().trim()).to.equal(''));
+    }
+};
+
+const CheckStatusUpdated = (exceptions_given, late_days_remaining) => {
+
+    for (const user_id of all_user_ids) {
+        cy.login(user_id);
+        cy.visit(['sample', 'late_table']);
+        // Wait for login change to take place
+        const status = predictedStatus(1 ,Math.max(0,all_late_users[user_id]-exceptions_given), late_days_remaining);
+
+        // Find late day status within the row in the late day usage table
+        cy.get('td[data-before-content="Event/Assignment"]')
+            .contains('Late Allowed Homework')
+            .siblings('td[data-before-content="Status"]')
+            .contains(status)
+            .should('exist');
+
+        cy.logout();
+    }
+};
+// Object with user_ids that have late submissions for gradeables
+const all_late_users = {}; // {user_id: #days_late}
+const all_user_ids = [];
+
+all_late_users['moscie'] = 3;
+all_user_ids.push('moscie');
+// Submission is 3 days late and 0 late days => Bad (too many late days used this term)
+// After given 2 late days => Bad (too many late days used this term)
+// Or After given 2 extentions => Bad (too many late days used this term)
+all_late_users['barteh'] = 2;
+all_user_ids.push('barteh');
+// Submission is 2 days late and 0 late days => Bad (too many late days used this term)
+// After given 2 late days => Bad (too many late days used on this assignment) because only 1 late day is allowed
+// Or After given 2 extentions => Good
+all_late_users['harbel'] = 1;
+all_user_ids.push('harbel');
+// Submission is 1 day late and 0 late days => Bad (too many late days used this term)
+// After given 2 late days => Late (valid submission)
+// Or After given 2 extentions => Good
+
+describe('Test cases involving late day cache updates', () => {
+    // Ignore uncaught js exceptions
+    Cypress.on('uncaught:exception', () => {
+        return false;
+    });
+
+    describe('Test accessing Bulk Late Days page as a student', () => {
+        it('should not allow access', () => {
+            cy.visit(['sample', 'bulk_late_days']);
+            cy.login('student');
+            cy.get('.content').contains("You don't have access to this page");
+        });
+    });
+
+    describe('Test accessing Bulk Late Days as an instructor', () => {
+        it('should load properly', () => {
+            cy.visit(['sample', 'bulk_late_days']);
+            cy.login('instructor');
+            cy.get('#late-day-table');
```

Comment:
```suggestion
            cy.get('#late-day-table').should('be.visible'); 
```

Or .should('exist'); which ever one will work.